src/core/lib/channel/channelz.cc

```diff
@@ -434,12 +436,15 @@ void PopulateSocketAddressJson(Json::Object* json, const char* name,
     if (!port.empty()) {
       port_num = atoi(port.data());
     }
-    char* b64_host = grpc_base64_encode(host.data(), host.size(), false, false);
+    grpc_resolved_address resolved_host;
+    grpc_string_to_sockaddr(&resolved_host, host.c_str(), port_num);
+    std::string packed_host = grpc_sockaddr_get_packed_host(&resolved_host);
+    std::string b64_host;
+    absl::Base64Escape(packed_host, &b64_host);
```

Comment:
```suggestion
    std::string b64_host = absl::Base64Escape(packed_host);
```

Reply:
works either way I guess

Reply:
Committable suggestions? Very nifty.

Yeah, I didn't notice that there was a simpler signature. Agree that yours is better.

Reply:
Yeah, that's the first time I tried out commitable suggestions. Seems very convenient for minor fixes.